linux/LinuxProcessList.c

```diff
@@ -1605,5 +1626,13 @@ void ProcessList_goThroughEntries(ProcessList* super, bool pauseProcessUpdate) {
 
    struct timeval tv;
    gettimeofday(&tv, NULL);
-   LinuxProcessList_recurseProcTree(this, PROCDIR, NULL, period, tv);
+   unsigned long long now = tv.tv_sec * 1000ULL + tv.tv_usec / 1000ULL;
+
+   int procDirfd = open(PROCDIR, O_RDONLY | O_DIRECTORY | O_NOFOLLOW);
+   if (procDirfd < 0) {
+      CRT_fatalError("Can not open the root process directory " PROCDIR);
+   }
+
+   /* LinuxProcessList_recurseProcTree() closes procDirfd */
+   LinuxProcessList_recurseProcTree(this, procDirfd, NULL, period, now);
```

Comment:
Not quite happy with this design. IMHO a resource provided by the caller should be freed by the caller. Makes for a few more LOC, but is cleaner from a resource management PoV IMO.

Reply:
reworked